troposphere/static/js/components/common/InstanceDetail.jsx

```diff
@@ -102,15 +118,33 @@ const InstanceDetail = React.createClass({
     },
 
     render: function() {
-        let { params } = this.props;
-        let { InstanceStore, InstanceHistoryStore } = this.props.subscriptions;
-        let instances = InstanceStore.getAll();
-        let instance = instances && instances.get(params.id);
-        let history = InstanceHistoryStore.fetchWhere({
-            "instance": params.id
-        })
-        if (!history || !instances) {
-            return <div className="loading" />
+        let { instance } = this.props,
+            { InstanceStore, InstanceHistoryStore } = this.props.subscriptions,
+            instances = InstanceStore.getAll(),
+            history = null;
+
+        // if we are provided w/ an instance,
+        // then fetch history using provided id in paramas
+        if (!instance) {
+            let { instanceId } = this.props.params;
+
+            history = InstanceHistoryStore.fetchWhere({
+                "instance": instanceId
```

Comment:
I think the BaseStore accounts for this (?) but just in case it doesn't, `instanceId` is going to be a string here, where as `instance.id` is going to be a number. Calling it out in case that's an issue, and we need to convert for consistency like `Number(instanceId)`.

Reply:
I believe that the `BaseStore` does handle this. But let me confirm this quick.

I think it's a String in both cases.

Reply:
So, I think these ultimately end up encoded into URLs as strings ... 

But, for the case when we are looking at `instanceId`, it is a string
```
$ this.props.params.instanceId
"35861"
```

And, when we look at the _idAttribute_ (`model.id`), we are getting a `Number`:
```
$ this.props.instance.id
35578
```

But I when working locally, there is a fetch and the expected results are returned.